homeassistant/components/plugwise/config_flow.py

```diff
@@ -154,6 +170,33 @@ def is_matching(self, other_flow: Self) -> bool:
 
         return False
 
+    async def _verify_connection(
+        self, user_input: dict[str, Any]
+    ) -> tuple[Smile | None, dict[str, str]]:
+        """Verify gateway connection helper function user and reconfiguration steps."""
+        errors: dict[str, str] = {}
+
+        try:
+            api = await validate_input(self.hass, user_input)
+        except ConnectionFailedError:
+            errors[CONF_BASE] = "cannot_connect"
+        except InvalidAuthentication:
+            errors[CONF_BASE] = "invalid_auth"
+        except InvalidSetupError:
+            errors[CONF_BASE] = "invalid_setup"
+        except (InvalidXMLError, ResponseError):
+            errors[CONF_BASE] = "response_error"
+        except UnsupportedDeviceError:
+            errors[CONF_BASE] = "unsupported"
+        except Exception:  # noqa: BLE001
+            errors[CONF_BASE] = "unknown"
+        else:
+            await self.async_set_unique_id(
+                api.smile_hostname or api.gateway_id, raise_on_progress=False
+            )
```

Comment:
This method is called `_verify_connection` but actually does a bit more than that. Should we move `async_set_unique_id` out to reduce the scope of this method?

Reply:
I moved it in because both user and reconfigure needed it (as potential de-duplication)

Reply:
Personally, I would prefer duplication and keeping this method simpler, because someone reading the method name would never assume that this method sets the unique id.
Alternatively, you could also have an `_async_set_unique_id(api)` method that does this, and call it from both user and reconfig steps.

Reply:
Tnx, I think I would duplicate it then

Reply:
And adjusted for mypy